[JS] Topic: DOM in JavaScript

manishkumar-hub opened the pull request:
<hr>

## Description 📜

I have created a documentation on DOM in javascript.
Fixes #2103 

<hr>

## Type of change 📝

<!----Please delete the hashtag from the correct option----->

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x ] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

<!----Please delete the hashtag from your domain----->

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ x] JavaScript
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

<!----Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this----->

- [x ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x ] I have performed a self-review of my own code or work.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x ] My changes generates no new warnings.
- [ x] I'm GWOC'21 contributor

<hr>

<!----Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this----->

## Screenshots / Gif (Optional) 📸

<hr>

paramsiddharth reviewed (changes requested):
Please talk about the following topics:
- `document.body`
- `document.forms`
- `name` attribute
- `document.body`
- `document.head`

paramsiddharth commented:
@Creatoon @dakshsethi @s4-2 Please review.

manishkumar-hub commented:
Reviewed and added about following topics :

- document.head
-  document.body
- document.forms
- name attribute

manishkumar-hub commented:
@paramsiddharth  Sir , Please review

paramsiddharth commented:
@manishkumar-hub Are you sure you added information about the `name` attribute in HTML? Please verify and let me know.

Creatoon commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,140 @@
+
+
+## What is DOM ?
+ DOM stands for Document object model . It is an application programming interface for manipulating HTML and XML documents. The DOM represents a document as a tree of nodes . It allows programs and scripts to dynamically access and update the content, structure, and style of a document.
```
```suggestion
 DOM stands for Document Object Model . It's an API (application programming interface) for manipulating HTML and XML documents. The DOM represents a document as a tree of nodes . It allows programs and scripts to dynamically access and update the content, structure, and style of a document.
```

Creatoon commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,140 @@
+![DOM ](https://searchengineland.com/figz/wp-content/seloads/2015/05/Microsoft-PowerPoint.jpg)
+
+
+## Use of DOM in javascript 
```
```suggestion
## Use of DOM in JavaScript
```

Creatoon commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,140 @@
+* JavaScript can create new HTML events in the page
+
+
+## Document object in JS
```
```suggestion
## Document object in JavaScript
```
whatever you write JS or javascript, write them in a same format, these things doesn't matters at all, but for a clean doc it really matters.

manishkumar-hub commented:
> @manishkumar-hub Are you sure you added information about the `name` attribute in HTML? Please verify and let me know.

That time i added only the " document.getElementsByName(name) " method  , but now i have added the detailed use of name attribute in forms . Do i need to add more about it or is it fine now?

paramsiddharth commented:
> > @manishkumar-hub Are you sure you added information about the `name` attribute in HTML? Please verify and let me know.
> 
> That time i added only the " document.getElementsByName(name) " method , but now i have added the detailed use of name attribute in forms . Do i need to add more about it or is it fine now?

Correct! Good job, Manish.
@Creatoon @dakshsethi @s4-2 Please review.

s4-2 reviewed (approved):
Looks good to me. Well done @manishkumar-hub

s4-2 reviewed (commented):
@manishkumar-hub  Just please add your name at the last line of the file as
**__Contributor__: [Your name](your github link)**

s4-2 reviewed (commented):
Please do require changes @manishkumar-hub

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+
+
+## What is DOM ?
+ DOM stands for Document Object Model . It's an API (application programming interface) for manipulating HTML and XML documents. The DOM represents a document as a tree of nodes . It allows programs and scripts to dynamically access and update the content, structure, and style of a document.
```
```suggestion
 DOM stands for **Document Object Model**. It's an API (application programming interface) for manipulating HTML and XML documents. The DOM represents a document as a tree of nodes. It allows programs and scripts to dynamically access and update the content, structure, and style of a document.
```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+
+## Use of DOM in JavaScript
+<br>
+With the DOM , javascript get all the power it needs to create dynamic webpage . With DOM :
```
```suggestion
With the DOM, JavaScript gets all the power it needs to create dynamic web pages. With the DOM :
```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+* JavaScript can react to all existing HTML events in the page
+
+* JavaScript can create new HTML events in the page
+
```
Write like this, without leaving extra lines

```
- JavaScript can change all the HTML elements in the page
- JavaScript can change all the HTML attributes in the page
- JavaScript can change all the CSS styles in the page
- JavaScript can remove existing HTML elements and attributes 
- JavaScript can add new HTML elements and attributes
- JavaScript can react to all existing HTML events in the page
- JavaScript can create new HTML events in the page
```

This will show up like the same only, less empty lines will be there... (Preview of above code)
- JavaScript can change all the HTML elements in the page
- JavaScript can change all the HTML attributes in the page
- JavaScript can change all the CSS styles in the page
- JavaScript can remove existing HTML elements and attributes 
- JavaScript can add new HTML elements and attributes
- JavaScript can react to all existing HTML events in the page
- JavaScript can create new HTML events in the page

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+
+## Document object in JavaScript
+<br>
+
```
Don't leave extra empty lines...

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+### HTML | DOM head Property
+
+<br>
+The head property returns the < head> element of the current document and in case if if there are more than one < head> element in the document, this property returns the first one.
```
``` suggestion
The head property returns the `<head>` element of the current document and in case if there are more than one `<head>` element in the document, this property returns the first one.
 ```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+
+### HTML | DOM body Property
+<br>
+It  returns the content present in the < body> tag. This property is used to view or change the  content present inside the < body> element and sets them with the new specified content.
```
```suggestion
It returns the content present in the `<body>` tag. This property is used to view or change the content present inside the `<body>` element and sets them with the new specified content.
```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+
+### HTML | DOM forms Collection
+<br>
+The forms collection returns a collection of all < form> elements present in the document . Since there may be multiple forms in a page , the elements are sorted as they are present in the HTML source code . 
```
```suggestion
The forms collection returns a collection of all `<form>` elements present in the document. Since there may be multiple forms on a page, the elements are sorted as they are present in the HTML source code. 
```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
````diff
@@ -0,0 +1,155 @@
+
+ ```
+
+  The name attribute which specifies the name of an < input > element is used to reference form elements in a JavaScript, or to reference form data after a form is submitted.
````
```suggestion
The name attribute which specifies the name of an `<input>` element is used to reference form elements in JavaScript, or to reference form data after a form is submitted.
```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+It is important to note that only form elements with a name attribute will have their values passed while submitting a form.
+
+<br>
+To access any particular element , DOM provides various method . Some of them are as follows :
```
```suggestion
To access any particular element, DOM provides various methods. Some of them are as follows :
```

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+| document.getElementsByClassName(name)  | 	Find elements by class name  |
+
+<br>
+<br>
```
Remove these 2 `<br>` tags
So much empty space is there.
1 empty line is enough.

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+<br>
+DOM provides various methods to add and delete HTML elements via javascript . Some of them are as follows :
+<br>
+<br>
```
Same here...
Pls work on it.

manishkumar-hub replied:
Thank you sir for your valuable suggestions . I have done all the required changes.

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
```diff
@@ -0,0 +1,155 @@
+
+<br>
+
+This was a brief  note about DOM .Above methods are frequently used methods , apart from them DOM contains much more methods for manipulating HTML elements . To refer more about DOM , Please visit : [DOM](https://www.w3schools.com/js/js_htmldom_document.asp) 
```
Add this line at the end
**Contributor**: [Manish Kumar Pandit](https://github.com/manishkumar-hub)
`**Contributor**: [Manish Kumar Pandit](https://github.com/manishkumar-hub)`

dakshsethi reviewed (changes requested):
Kindly work on this document.
Especially there are soo many empty lines!!

manishkumar-hub commented:
I have done the required changes . @dakshsethi @Creatoon Please review .

dakshsethi commented on JavaScript/Topics/13.DOM_in_javascript/README.md:
````diff
@@ -0,0 +1,116 @@
+
+The forms collection returns a collection of all `<form>` elements present in the document. Since there may be multiple forms on a page, the elements are sorted as they are present in the HTML source code. <br>
+E.g .
+ ```
````
Since this is a JavaScript code block, write `js` after ` ``` `
Do these changes everywhere where there is a JavaScript code block
```suggestion
 ```js
```

dakshsethi reviewed (approved):
Looks good to me!! 🔥🔥🔥

manishkumar-hub commented:
@Creatoon  sir, Please review.
